lfr/heterogeneous_single_assembly_3D/neutronics_standalone/neutronics.i

```diff
@@ -547,1341 +547,1341 @@ totalpower = 3700000.0 # W
-       AM241:2.087265E-04
-       O16:4.444138E-02'
+       U234 1.769055E-07
+       U235 4.404137E-05
+       U238 1.735054E-02
+       PU238 1.244939E-05
+       PU239 3.413306E-03
+       PU240 1.319941E-03
+       PU241 8.656869E-05
+       PU242 1.234538E-04
+       AM241 2.087265E-04
+       O16 4.444138E-02'
   []
   [Fuel_Ring7]
-    type = MicroNeutronicsMaterial
+    type = IsotopeComposition
     block = '${bid_F_fuel_R7}'
-    library_id = '${lid_F_fuel_R7}'
-    materials = '
+    composition_ids = '${lid_F_fuel_R7}'
+    isotope_densities = '
      Fuel_Ring7 ${mid_F_fuel_R7}
-       U234:1.769055E-07
-       U235:4.404137E-05
-       U238:1.735054E-02
-       PU238:1.244939E-05
-       PU239:3.413306E-03
-       PU240:1.319941E-03
-       PU241:8.656869E-05
-       PU242:1.234538E-04
-       AM241:2.087265E-04
-       O16:4.444138E-02'
+       U234 1.769055E-07
+       U235 4.404137E-05
+       U238 1.735054E-02
+       PU238 1.244939E-05
+       PU239 3.413306E-03
+       PU240 1.319941E-03
+       PU241 8.656869E-05
+       PU242 1.234538E-04
+       AM241 2.087265E-04
+       O16 4.444138E-02'
   []
   [Fuel_Clad]
-    type = MicroNeutronicsMaterial
+    type = IsotopeComposition
     block = '${bid_F_clad}'
-    library_id = '${lid_F_clad}'
-    materials = '
+    composition_ids = '${lid_F_clad}'
+    isotope_densities = '
      Fuel_Clad ${mid_F_clad}
-       FE54:3.185952E-03
-       FE56:5.001168E-02
-       FE57:1.154947E-03
-       FE58:1.537129E-04
-       NI58:8.373412E-03
-       NI60:3.225451E-03
-       NI61:1.402235E-04
-       NI62:4.469793E-04
-       NI64:1.138947E-04
-       CR50:5.643439E-04
-       CR52:1.088250E-02
-       CR53:1.234043E-03
-       CR54:3.071758E-04
-       MN55:1.271641E-03
-       MO92:1.080550E-04
-       MO94:6.735188E-05
-       MO95:1.159146E-04
-       MO96:1.214544E-04
-       MO97:6.953578E-05
-       MO98:1.757019E-04
-       MO100:7.011875E-05
-       SI28:1.300140E-03
-       SI29:6.601594E-05
-       SI30:4.351798E-05
-       C:3.489938E-04
-       P31:6.766687E-05
-       S32:2.068704E-05
-       S33:1.656123E-07
-       S34:9.348567E-07
-       S36:4.358298E-09
-       TI46:3.210951E-05
-       TI47:2.895766E-05
-       TI48:2.869267E-04
-       TI49:2.105602E-05
-       TI50:2.016107E-05
-       V:2.742873E-05
-       ZR90:7.880535E-06
-       ZR91:1.718520E-06
-       ZR92:2.626878E-06
-       ZR94:2.662077E-06
-       ZR96:4.288701E-07
-       W182:2.014107E-06
-       W183:1.087650E-06
-       W184:2.337892E-06
-       W186:2.160800E-06
-       CU63:1.520930E-05
-       CU65:6.778986E-06
-       CO59:2.370990E-05
-       CA40:3.379743E-05
-       CA42:2.255696E-07
-       CA43:4.706582E-08
-       CA44:7.272563E-07
-       CA46:1.394535E-09
-       CA48:6.519498E-08
-       NB93:7.519752E-06
-       N14:4.969370E-05
-       N15:1.835515E-07
-       AL27:2.589280E-05
-       TA181:3.861021E-06
-       B10:5.144462E-06
-       B11:2.070704E-05'
+       FE54 3.185952E-03
```

Comment:
@YaqiWang how do I fix this?
FE54 is present in two 'sub-compositions' in that map (it s not a thing maybe) so the code complains of a duplicate

Reply:
@GiudGiud each composition should be containing only one set of isotopes. The materials of the old input have multiple compositions, you will need to add multiple compositions for each material. I think this input was created by Hansol at ANL. You can ask some advices from him on griffin slack channel. I am not sure whether all materials have different extra element ids.

Reply:
@pak1sol do you have a script to merge these compositions together?